### PR TITLE
check and fix packages urls

### PR DESCRIPTION
**Pull request by drodin**
* script to go through packages and check if urls are still valid
* manual clean up of non existent/changed urls
* some cosmetic fixes for easier automation checks on packages

This fixes: https://github.com/cpp-pm/hunter/issues/546 https://github.com/cpp-pm/hunter/issues/540 and https://github.com/cpp-pm/hunter/issues/191

### CODE REVIEW

**Review by NeroBurner — CHANGES_REQUESTED** [review]
Thank you very much for this contribution! Checking the download URLs is a VERY good idea!

I'd like to extend the script a bit. Maybe it would be a good idea to check the projects when they are changed in the CI. On the one hand checking the project URL when a contributor comes along and gets a seemingly unconnected error because of an old package URL. Otherwise it would be great to check the URLs somewhere/somewhen

@rbsheth what do you think?

File: maintenance/check-urls.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        return c.getinfo(pycurl.HTTP_CODE)
+    except pycurl.error:
+        return 999
+
+hunterDir = dirname(dirname(abspath(__file__)))
+projectsDir = join(hunterDir, 'cmake', 'projects')
+
+projectsFiles = join(projectsDir, '**', '*.cmake')
+
+checkedFile = join(hunterDir, 'checked.txt')
```
[the remark this answers is not in the document]
Comment (drodin) [reply maintenance/check-urls.py:25]: done https://github.com/cpp-pm/hunter/pull/575/commits/9698dd773f672e069075f48b3761492e640c9820

File: maintenance/check-urls.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        c.setopt(pycurl.FOLLOWLOCATION, 1)
+        c.perform()
+        return c.getinfo(pycurl.HTTP_CODE)
+    except pycurl.error:
+        return 999
+
+hunterDir = dirname(dirname(abspath(__file__)))
+projectsDir = join(hunterDir, 'cmake', 'projects')
+
+projectsFiles = join(projectsDir, '**', '*.cmake')
```
[the remark this answers is not in the document]
Comment (drodin) [reply maintenance/check-urls.py]: https://github.com/cpp-pm/hunter/pull/575/commits/ac61f83550672e517ef0383d2a0d9d2488d04c13

**Review by NeroBurner — APPROVED** [review]
LGTM, only question left for me is if and how we can include this marvelous piece of code in the CI :thinking:

**Comment by drodin** [comment]
> LGTM, only question left for me is if and how we can include this marvelous piece of code in the CI 🤔

its possible to include this script into build.sh/cmd. add an exit code to it, or just check  if 'checked.txt' file is present and zero size

**Comment by rbsheth** [comment]
> Thank you very much for this contribution! Checking the download URLs is a VERY good idea!
> 
> I'd like to extend the script a bit. Maybe it would be a good idea to check the projects when they are changed in the CI. On the one hand checking the project URL when a contributor comes along and gets a seemingly unconnected error because of an old package URL. Otherwise it would be great to check the URLs somewhere/somewhen
> 
> @rbsheth what do you think?

Yeah, it would be useful to weed out broken packages. We can add it to CI

**Comment by NeroBurner** [comment]
created a new Issue for the adding of the script to the CI

https://github.com/cpp-pm/hunter/issues/598

**Comment by NeroBurner** [comment]
Released: https://github.com/cpp-pm/hunter/releases/tag/v0.24.4

**Pull request merged**